lib/Runtime/Debug/TTEventLog.cpp

```diff
@@ -2526,8 +2526,12 @@ namespace TTD
 
     void EventLog::InnerLoopEmitLog(const TTDebuggerSourceLocation& writeLocation, const char* emitUri, size_t emitUriLength)
     {
-        NSLogEvents::TTDInnerLoopLogWriteEventLogEntry* evt = nullptr;
-        this->RecordGetInitializedEvent<NSLogEvents::TTDInnerLoopLogWriteEventLogEntry, NSLogEvents::EventKind::TTDInnerLoopLogWriteTag>(&evt);
+        byte buff[TTD_EVENT_PLUS_DATA_SIZE(NSLogEvents::TTDInnerLoopLogWriteEventLogEntry)];
```

Comment:
not clear why you're allocating a byte[] and then casting to EventLogEntry, instead of just allocating an EventLogEntry on stack.   Latter would be more clear.

Reply:
The log layout is a packed setup where we expect the event payload data to immediately follow the event header. To ensure this layout is done correctly here we do the byte[] + manual casts instead of relying on the compiler to do contiguous layout on the stack. I'll add a comment here + ref to the other relevant code.